app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CartsController < ApplicationController
+  before_action :load_book, :check_quantity, only: :create
+
+  def create
+    if @quantity == false
+      return respond_to do |format|
+        format.js
+      end
+    end
+    add_book_to_cart(params[:book_id], params[:quantity])
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  private
+
+  def check_quantity
+    return if @book.quantity >= params[:quantity].to_i
+
+    @quantity = false
```

Comment:
cơ mà chị nghĩ chỉ cần 
```suggestion
    @quantity = @book.quantity >= params[:quantity].to_i
```

ở line 5, thì ` unless check_quantity...` và ko cần gọi action filter cho method check_quantity